torch/_ops.py

```diff
@@ -337,6 +337,9 @@ def check_overloaded(arg):
                         # "natural" calling convention: (mode, *args, **kwargs)
                         # TODO(rzou): we should support torch_dispatch calling convention too.
                         result = handler(mode, *args, **kwargs)
+                elif type(curr_mode) is torch._subclasses.FakeTensorMode:
+                    with _pop_mode_temporarily() as mode:
+                        result = mode.__torch_dispatch__(self, [], args, kwargs)
```

Comment:
Alternative: have `@register_hop_fake(invoke_subgraph)` also call 
```
@invoke_subgraph.py_impl(FakeTensorMode)
def _(mode, *args, **kwargs):
    return mode.__torch_dispatch__(self, [], args, kwargs)
```
so we don't need to special case it here. What do you think?

Reply:
Very good idea. Implemented.